clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -414,6 +419,66 @@ public void testCreatePartitions() throws Exception {
         }
     }
 
+    @Test
+    public void testDeleteRecords() throws Exception {
+
+        HashMap<Integer, Node> nodes = new HashMap<>();
+        nodes.put(0, new Node(0, "localhost", 8121));
+        List<PartitionInfo> partitionInfos = new ArrayList<>();
```

Comment:
@guozhangwang if I understood your suggestion it sounds to me to change the `mockClientEnv` signature in this way : 

`mockClientEnv(Collection<PartitionInfo> partitions, String... configVals)`

having all the current tests which don't need it, passing just `Collections.<PartitionInfo>emptySet()` while the current delete records test builds the collection and then pass it to the method. Is this what were you thinking ?

Reply:
@guozhangwang if I understood your suggestion it sounds to me to change the `mockClientEnv` signature in this way :

`mockClientEnv(Collection<PartitionInfo> partitions, String... configVals)`

having all the current tests, which don't need it, passing just `Collections.<PartitionInfo>emptySet()` while the current delete records test builds the collection and then pass it to the method. Is this what were you thinking ?

Reply:
Yes that was my suggestion.

Reply:
I have taken a look to the other tests. All of them use just the `mockClientEnv` method. Only two tests build the cluster from scratch : `testDeleteRecords` and `testHandleTimeout`.
I can change the `mockClientEnv` for using it in the `testDeleteRecords` but we still have the problem for the `testHandleTimeout`. Maybe we should leave the `mockClientEnv` as it is because it works for the most of the tests and handling specific use cases as we do today ?

Reply:
Yeah I think this is a subjective opinion. I'm fine with it staying as is.